hotsos/core/plugintools.py

```diff
@@ -224,6 +247,113 @@ def bind_interfaces(self):
         """
 
 
+def summary_entry(name, index=0):
+    """Decorate a class member function to indicate that it is a summary
+    function. Summary functions are automatically discovered by the
+    PluginPartBase class and will be part of the summary output."""
+
+    def real_decorator(func):
+        # Store the summary metadata information on the function itself.
+        func.__summary_elem_name = name
+        func.__summary_elem_index = index
+        return func
+    return real_decorator
+
+
+def get_min_available_entry_index():
+    """ Return first index available after default entries. """
+    return DefaultSummaryEntryIndexes.available
+
+
+@dataclass(frozen=True)
+class DefaultSummaryEntryIndexes():
+    """ Indexes used for default summary entries. """
+    version: str = 0
+    release: str = 1
+    services: str = 2
+    snaps: str = 3
+    dpkg: str = 4
+    docker_images: str = 5
+    # The following index represents the first one that is available for
+    # implementations to use and they implement their indexes as an offset from
+    # this value so that if it is increased they are automatically increased.
+    available: str = 6
+
+
+class SummaryBase(ApplicationBase):
+    """ Common structure for application summary output.
+
+    Individual application plugins should implement this class and extend to
+    include information specific to their application.
+    """
+    entry_index_version = 0
```

Comment:
I haven't seen these used. What's the purpose?

Reply:
ah sorry yeah i forgot to delete these when i moved them to DefaultSummaryEntryIndexes

Reply:
fixed.